src/main/jbake/content/blog/2017/12/running-agent.adoc

```diff
@@ -0,0 +1,155 @@
+= Running Hawkular Agent
+John Mazzitelli
+2017-12-11
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, agent, prometheus, metrics
+
+Now that Hawkular is moving towards integraton with Prometheus as its metrics collection and storage system, the agent
+has had some changes. This document will discuss those changes and how to run the Hawkular Agent.
+
+First, the Hawkular Agent now only runs as a javaagent (no longer does it run inside WildFly as a subsystem extension).
+So to run the agent, simply attach it as a javaagent to whatever JVM you want to monitor (be it a WildFly or EAP app server or
+any JVM application that exposes metrics via JMX). For example, add this command line option to the command that starts the JVM:
+
+```
+-javaagent:hawkular-javaagent.jar=config=hawkular-javaagent-config.yaml,delay=10
+```
+
+The configuration file contains all the settings necessary to start the agent and connect to the Hawkular Server.
+Once connected, the agent will pull down additional configuration settings from the server.
+
+=== Running in WildFly
+
+If you are attaching the agent to a WildFly server, you must set some additional settings.
+
+If you are running WildFly in standalone mode, the standalone.conf you should have something like:
+
+```
+JBOSS_MODULES_SYSTEM_PKGS="org.jboss.byteman,org.jboss.logmanager"
+...
+JAVA_OPTS="$JAVA_OPTS -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
+...
+JAVA_OPTS="$JAVA_OPTS -javaagent:$JBOSS_HOME/bin/hawkular-javaagent.jar=config=$JBOSS_HOME/standalone/configuration/hawkular-javaagent-config.yaml,delay=10"
+
+```
+
+If you are running WildFly in domain mode, the domain.conf you should have something like:
```

Comment:
Is the domain.conf file on the host controller?  Would it be helpful to include that information for new users?

Reply:
I am assuming if the user is running domain mode, they know this (even if they are new users to hawkular, if they know what domain mode is, they know the domain.conf).

But I will reword to say: "If you are running WildFly in domain mode, the host controller's domain.conf file should have something like:"

Reply:
We should be careful about assuming what users already know.  At some point everyone is a new user and they're reading the docs because they DON'T know.